internal/pkg/otel/manager/healthcheck.go

```diff
@@ -0,0 +1,231 @@
+func (e *healthCheckEvent) Timestamp() time.Time           { return e.timestamp }
+func (e *healthCheckEvent) Err() error                     { return e.err }
+
+// AllComponentsStatuses retrieves the status of all components from the health check endpoint.
+func AllComponentsStatuses(ctx context.Context, httpHealthCheckPort int) (*status.AggregateStatus, error) {
+	ctx, cancel := context.WithTimeout(ctx, 5*time.Second)
+	defer cancel()
+
+	req, err := http.NewRequestWithContext(ctx, "GET", fmt.Sprintf("http://localhost:%d/%s?verbose",
+		httpHealthCheckPort, healthCheckHealthStatusPath), nil)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create http request: %w", err)
+	}
+
+	client := http.DefaultClient
+	res, err := client.Do(req)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get status: %w", err)
+	}
+	defer res.Body.Close()
+
+	body, err := io.ReadAll(res.Body)
+	if err != nil {
+		return nil, fmt.Errorf("failed to read body: %w", err)
+	}
+
+	serStatus := &SerializableStatus{}
+	err = json.Unmarshal(body, serStatus)
+	if err != nil {
+		return nil, fmt.Errorf("failed to unmarshal serializable status: %w", err)
+	}
+
+	return fromSerializableStatus(serStatus), nil
+}
+
+// fromSerializableStatus reconstructs an AggregateStatus from serializableStatus.
+func fromSerializableStatus(ss *SerializableStatus) *status.AggregateStatus {
+	ev := fromSerializableEvent(ss.SerializableEvent)
+
+	as := &status.AggregateStatus{
+		Event:              ev,
+		ComponentStatusMap: make(map[string]*status.AggregateStatus),
+	}
+
+	for k, cs := range ss.ComponentStatuses {
+		as.ComponentStatusMap[k] = fromSerializableStatus(cs)
+	}
+
+	return as
+}
+
+// fromSerializableEvent reconstructs a status.Event from SerializableEvent.
+func fromSerializableEvent(se *SerializableEvent) status.Event {
+	if se == nil {
+		return nil
+	}
+
+	var err error
+	if se.Error != "" {
+		err = errors.New(se.Error)
+	}
+
+	statusVal, ok := stringToStatusMap[se.StatusString]
+	if !ok {
+		statusVal = componentstatus.StatusNone
+	}
+
+	return &healthCheckEvent{
+		status:    statusVal,
+		timestamp: se.Timestamp,
+		err:       err,
+	}
+}
+
+// aggregateStatus creates a new AggregateStatus with the provided component status and error.
+func aggregateStatus(sts componentstatus.Status, err error) *status.AggregateStatus {
+	return &status.AggregateStatus{
+		Event: &healthCheckEvent{
+			status:    sts,
+			timestamp: time.Now(),
+			err:       err,
+		},
+		ComponentStatusMap: make(map[string]*status.AggregateStatus),
+	}
+}
+
+// injectHeathCheckV2Extension injects the healthcheckv2 extension into the provided configuration.
+func injectHeathCheckV2Extension(conf *confmap.Conf, httpHealthCheckPort int) error {
+	nsUUID, err := uuid.NewV4()
+	if err != nil {
+		return fmt.Errorf("cannot generate UUID V4: %w", err)
+	}
+	healthCheckExtensionID := fmt.Sprintf("healthcheckv2/%s", nsUUID.String())
```

Comment:
Not sure what the first part of the ID (the part to the left of the `/`) is supposed to mean but would it make sense to use the `healthCheckExtensionName` constant here?

Reply:
this is deliberate more context [here](https://github.com/elastic/elastic-agent/pull/8248#discussion_r2121076122) 🙂

Reply:
The idiomatic way to create this id would be:

```go
	componentType := component.MustNewType(healthCheckExtensionName)
	healthCheckExtensionID := component.NewIDWithName(componentType, nsUUID.String())
	err = conf.Merge(confmap.NewFromStringMap(map[string]interface{}{
		"extensions": map[string]interface{}{
			healthCheckExtensionID.String(): map[string]interface{}{
```

But this is a small nit.